### ISSUE TITLE
Query with collation not using index with collation

### ISSUE THREAD

**Issue by snej** [couchbase/couchbase-lite-core#249]
Running the TunesPerfTest, I'm finding that the second time the queries (queryAllArtists and queryAlbums) run, they aren't any faster, even though the index has been created. Explaining the queries shows that the SQLite query planner isn't using the index at all.

(Below I'm using the standard `NOCASE` collation instead of our custom Unicode one, just to try to eliminate some variables. It still fails.)

queryAlbums generates the SQL query:
```sql
SELECT sequence, key, version, flags, fl_value(body, 'Album') 
FROM kv_default 
WHERE fl_value(body, 'Artist') COLLATE NOCASE = $_ARTIST 
      AND fl_value(body, 'Compilation') IS NULL 
GROUP BY fl_value(body, 'Album') COLLATE NOCASE 
ORDER BY fl_value(body, 'Album') COLLATE NOCASE 
LIMIT ifnull($_x_limit, -1) OFFSET ifnull($_x_offset, 0)
```

Without the index this is explained as using a slow linear scan:
```
0|0|0| SCAN TABLE kv_default
0|0|0| USE TEMP B-TREE FOR GROUP BY
```

The index is created as:
```sql
CREATE INDEX "byArtist" ON kv_default (fl_value(body, 'Artist') COLLATE NOCASE,
                                       fl_value(body, 'Compilation'))
```

But afterwards the query is still using the slow scan:
```
0|0|0| SCAN TABLE kv_default
0|0|0| USE TEMP B-TREE FOR GROUP BY
```

What it _should_ be doing is a fast search of the index:
```
0|0|0| SEARCH TABLE kv_default USING INDEX byArtist (<expr>=? AND <expr>=?)
0|0|0| USE TEMP B-TREE FOR GROUP BY
```

**Comment by snej** [comment]
After struggling with this, I think it comes down to the version of SQLite. We're currently using 3.15.2 (according to SQLITE_VERSION in our sqlite3.h.) My Mac has 3.19.3 installed, and when I open the database with the `sqlite3` tool and have it explain the same query, it generates the correct/fast query plan.

**Comment by snej** [comment]
Verified that updating our statically-linked SQLite to 3.19 fixes the problem. But we need to get it properly integrated with the SQLCipher goodies, for it to be a real fix.

**Comment by snej** [comment]
SQLite 3.20 has a bug that causes crashes in LiteCore (see #259) so we can't use it. The fix will be in the upcoming 3.21 release. Until then we should stick with 3.19.

**Comment by snej** [comment]
Our SQLCipher submodule has been updated to SQLite 3.19.
